var/spack/repos/builtin/packages/tassel/package.py

```diff
@@ -13,6 +13,7 @@ class Tassel(Package):
     homepage = "https://www.maizegenetics.net/tassel"
     git      = "https://bitbucket.org/tasseladmin/tassel-5-standalone.git"
 
+    version('3.0.174', commit='612a92c0f677d6ec040c17e146671974a3cdc2da', git='git://git.code.sf.net/p/tassel/tassel3-standalone.git')
```

Comment:
Reading [here](https://bytebucket.org/tasseladmin/tassel-5-source/wiki/docs/UpdatingTasselStandaloneUsingGit.pdf) I think this package might need some reworking. I concur the version number you used is better, but then my understanding is that the `2017-07-22` version is in the 5.X series, right? Can we use consistent version numbers?

Reply:
Also, it seems to me that the constraints on `java` are not encoded in the `package.py`.

Reply:
the 2017-07-22 version was in the package.py before I touched it.  I just updated it in my latest PR update after tracking down that commit... 

the tassel package depends on java, what constraints are you referring to?

Reply:
> the tassel package depends on java, what constraints are you referring to?

In the link above there are mentions of the required Java versions which are not encoded in `package.py`